hdx_hwa/engine/data_insertion_transformer.py

```diff
@@ -0,0 +1,52 @@
+from datetime import datetime, timezone
+from dateutil.parser import parse as dateutil_parse
+from typing import Any, Dict, List
+
+
+class DataInsertionTransformer:
+    """
+    A class that prepares a row of data to be inserted into the database
+    """
+
+    def __init__(self, table_col_to_type: Dict[str, str], csv_columns: List[str]) -> None:
+        self.table_col_to_type = table_col_to_type
+        self.csv_column_name_to_index = {col: i for i, col in enumerate(csv_columns)}
+
+    def _transform_string_to_datetime(self, value: str) -> datetime:
+        value = value.strip()
+        if value:
+            timestamp = dateutil_parse(value)
+            if timestamp.tzinfo:
+                timestamp = timestamp.astimezone(timezone.utc).replace(tzinfo=None)
+            return timestamp
+        else:
+            return None
+
+    def _transform_string_to_type(self, value: str, type: str) -> Any:
```

Comment:
Yes. The CSVs are generated by `psql` and it likes 't' and 'f' for Booleans. For int and float columns the strings are getting converted automatically. 
I'll soon get rif of the datetime transformation as well. I thought it was needed, but it actually only needs to be transformed to `None` in case of an empty string.